nexus-repository-helm/src/main/java/org/sonatype/repository/helm/internal/hosted/HelmHostedFacetImpl.java

```diff
@@ -95,31 +94,33 @@ public Content get(final String path) {
   }
 
   @Override
+  @TransactionalStoreBlob
   public void upload(final String path,
                      final Payload payload,
                      final AssetKind assetKind) throws IOException
   {
-    checkNotNull(path);
-    checkNotNull(payload);
-
-    if (assetKind != HELM_PACKAGE) {
-      throw new IllegalArgumentException("Unsupported assetKind: " + assetKind);
-    }
-
     try (TempBlob tempBlob = facet(StorageFacet.class).createTempBlob(payload, HASH_ALGORITHMS)) {
-      storeChart(path, tempBlob, payload);
+      upload(path, tempBlob, payload, assetKind);
     }
   }
 
   @Override
-  public Asset upload(String path, TempBlob tempBlob,Payload payload) throws IOException {
+  public Asset upload(String path, TempBlob tempBlob, Payload payload, AssetKind assetKind) throws IOException {
+    if (assetKind != HELM_PACKAGE && assetKind != HELM_PROVENANCE) {
+      throw new IllegalArgumentException("Unsupported assetKind: " + assetKind);
+    }
     checkNotNull(path);
     checkNotNull(tempBlob);
 
     StorageTx tx = UnitOfWork.currentTx();
     Bucket bucket = tx.findBucket(getRepository());
-
-    Asset asset = createChartAsset(path, tx, bucket, tempBlob.get());
+    InputStream inputStream = tempBlob.get();
+    HelmAttributes attributes =
+        assetKind == HELM_PROVENANCE ?
```

Comment:
`assetKind == HELM_PROVENANCE`
      `? helmAttributeParser.getAttributesProvenanceFromInputStream(inputStream) `
      `: helmAttributeParser.getAttributesFromInputStream(inputStream);`

It seems a bit more readable

Reply:
fixed